### ISSUE TITLE
A large diff between syntok and segtok on Web of Science datasets

### ISSUE THREAD

**Issue by newtover** [fnl/syntok#3]
Hello, 

first of all, thank you for the work you are doing. It's really great. 

I have been using `segtok` for some time and only today noticed the NB about `syntok`. And just decided to see whether there are any differences between them. For that I downloaded the [Web of Science](https://data.mendeley.com/datasets/9rw3vkcfy4/6) data sets and generated a diff on the smallest set with something like:

```
(flair) vagrant@vagrant:~$ /usr/bin/time python -m syntok.segmenter WebOfScience/WOS5736/X.txt >syn.txt
40.80user 0.46system 0:41.49elapsed 99%CPU (0avgtext+0avgdata 51244maxresident)k
(flair) vagrant@vagrant:~$ /usr/bin/time python -m segtok.segmenter WebOfScience/WOS5736/X.txt >seg.txt
55.01user 0.17system 0:55.40elapsed 99%CPU (0avgtext+0avgdata 77284maxresident)k
diff -U 0 seg.txt syn.txt > diff.txt
```
It turned out, there are lots of differences and not in favor of `syntok`. I thought, you might be interested in looking at the diff. My suspicion is that the issues are rather similar, but one needs to look through the file and extract the specific issues. I put the diff into a [gist](https://gist.github.com/newtover/be2b7862d56de6d731bc4e45c9cf43ac). If I have time, I will try to help with it. Thank you.

**Comment by fnl** [comment]
Hi @newtover - foremost, thanks looking into this and sharing the collection - much appreciated!

I think some parts are OK on syntok's side (I prefer how it does the copyright examples) but yes, it errs on the side of under-splitting and in some cases clearly breaks (e.g., the p-values), overall making segtok work better on this data, I agree.

And yes, I haven't spent as much time tuning syntok against all kinds of text data as I should have, hoping for exactly what you just did to either happen (thanks!) or to have time myself for that. But syntok certainly is the future, as it is more efficient, and the tokenization is more consistent, and the API is far less confusing.

I hope over the next weeks I can make improvements to syntok and add your data as tests to the library and fix the cases; Pls. stay tuned. (Naturally, any PRs that keep the library in the green are equally welcome!)

**Comment by fnl** [comment]
Hi @newtover - I just committed a new version that fixes most of the issues you were seeing, I believe, as candidate for v.1.1.1

See d5e1bec3cbbe061bbfe3f8dd65869ccbec39c440

Could you take that for a "test drive" and let me know if the differences are now sufficiently minor, or if not, point me to any major cases that still would disturb you? Thanks!

**Comment by fnl** [comment]
Closing as I released updates for this with 1.1.1 (now on PyPi). Please create a new ticket if you still spot any problems.
